lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+
+/**
+ * Html file representation (needs for errors output).
+ *
+ * @name HtmlFile
+ */
+var HtmlFile = function(filename, source) {
+    this._filename = filename;
+    this._lines = source.split(/\r\n|\r|\n/);
+};
+
+HtmlFile.prototype = {
+    /**
+     * Returns source filename for this object representation.
+     *
+     * @returns {String}
+     */
+    getFilename: function() {
+        return this._filename;
+    },
+    /**
+     * Returns array of source lines for the file.
+     *
+     * @returns {String[]}
+     */
+    getLines: function() {
+        return this._lines;
+    }
+};
+
+/**
+ * Parse html and retrieve script sources.
+ * @param {String} html
+ * @returns {Object[]}
+ */
+function getScripts(html) {
+    function onopen(name, attrs) {
+        // tag should be a <script>
+        if (name !== 'script') {
+            return;
+        }
+
+        // ignore scripts with src attribute
+        if (attrs.src) {
+            return;
+        }
+
+        // script tag should has no type attribute or attribute should be equal to text/javascript
+        if (attrs.type && attrs.type.toLowerCase() !== 'text/javascript') {
+            return;
+        }
+
+        // store script content start pos
+        scriptStartPos = parser.endIndex + 1;
+    }
+
+    function onclose() {
+        // if there is a script start pos
+        if (scriptStartPos) {
+            // get script content
+            var scriptEndPos = parser.startIndex;
+            var source = html.substring(scriptStartPos, scriptEndPos);
+
+            // store script content only if it contains non-whitespace characters
+            if (/\S/.test(source)) {
+                scripts.push({
+                    source: source,
+                    start: scriptStartPos,
+                    end: scriptEndPos
+                });
+            }
+
+            // reset script start position
+            scriptStartPos = 0;
+        }
+    }
+
+    var scriptStartPos = 0;
+    var scripts = [];
+    var parser = new htmlparser.Parser({
+        onopentag: onopen,
+        onclosetag: onclose
+    });
+
+    parser.parseComplete(html);
+
+    return scripts;
+}
+
+/**
+ * Trim script source and normalize lines offset.
```

Comment:
Nope, but it's easy to make ;-) https://github.com/jscs-dev/jscs-jsdoc/issues/85